platform-sdk/platform-apps/tests/ISSTestingTool/src/main/java/com/swirlds/demo/iss/ISSTestingToolState.java

```diff
@@ -236,6 +238,18 @@ <T extends SelfSerializable> void writeObjectByChildIndex(int index, List<T> lis
         }
     }
 
+    @Override
+    public void preHandle(
+            @NonNull Event event,
+            @NonNull Consumer<ScopedSystemTransaction<StateSignatureTransaction>> stateSignatureTransaction) {
+        event.forEachTransaction(transaction -> {
+            if (!transaction.isSystem() && areTransactionBytesSystemOnes(transaction)) {
+                stateSignatureTransaction.accept(
```

Comment:
I think we should accept a `StateSignatureTransaction`, that is converted from a `Transaction`, so that we load the correct type into the callback.